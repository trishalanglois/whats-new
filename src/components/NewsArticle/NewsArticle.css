.news-article {
  text-align: center;
  -webkit-column-break-inside: avoid;
  page-break-inside: avoid;
  break-inside: avoid;
  margin-bottom: 2em;
  border: 2px solid #fff;
}

img {
  width: 100%;
  height: 200px;
  object-fit: cover;
}

article {
  background-color: #fff;
}

h1, p {
  color: #000;
  margin: 1em;
}

a {
  color: #A115F1;
  font-size: 1.2em;
  border-top: 1px solid #000;
  display: block;
}

a:hover {
  font-weight: bold;
}
