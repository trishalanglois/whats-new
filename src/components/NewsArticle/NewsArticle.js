import React from 'react';
import './NewsArticle.css';

const NewsArticle = () => (
  <div className = 'news-article'>
    This is the news article.
  </div>
);
export default NewsArticle;
