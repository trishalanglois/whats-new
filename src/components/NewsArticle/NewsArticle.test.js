import React from 'react';
import NewsArticle from './NewsArticle';
import { shallow } from 'enzyme';

describe('NewsArtcile', () => {
  it('should correctly match the snapshot with all the data passed correctly', () => {
    const wrapper = shallow( <NewsArticle
      articles={[{
        image:'image path',
        headline: 'b',
        description:'b',
        url:'url'
      }, {
        image:'image path',
        headline: 'b',
        description:'b',
        url:'url'
      }]}
    />)
    expect(wrapper).toMatchSnapshot();
  })
})
