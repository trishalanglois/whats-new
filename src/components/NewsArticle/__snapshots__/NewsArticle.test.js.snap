// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArtcile should correctly match the snapshot with all the data passed correctly 1`] = `
<article
  className="news-article"
>
  <img />
  <h1 />
  <p />
  <a>
    Link to Article
  </a>
</article>
`;
